### PR TITLE
Improve integration tests

### PR DESCRIPTION
**Pull request by batiati**
While investigating this [Rafik issue](https://github.com/interledger/rafiki/actions/runs/7653128123/job/20854361776?pr=2364), I noticed that our Node.js CI passes even if tests fail.

- Ensure that the Node.js test runner returns a non-zero exit code when tests fail.
- Add Node, Java, and Dotnet tests for `pending_transfer_expired`.

### CODE REVIEW

File: src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java [hunk 1]
```diff
@@ -699,6 +699,103 @@ public void testCreatePendingTransfersAndVoid() throws Throwable {
         }
     }
 
+    @Test
+    public void testCreatePendingTransfersAndVoidExpired() throws Throwable {
+
+        try (var server = new Server()) {
```
Comment (matklad) [inline src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java]: Drive-by, but can’t we take advantage of using declaration here and save an indent?

Comment (matklad) [reply src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java]: 🤦  this is Java, not C#

Comment (matklad) [reply src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java]: Though, we should be able to put both the server and the client into a single try?

```
try (FileReader fr = new FileReader(path);
	         BufferedReader br = new BufferedReader(fr)) {
	        return br.readLine();
	    }
```

Comment (matklad) [reply src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java]: And I think we don’t need catch here?

File: src/clients/node/src/test.ts [hunk 2]
```diff
@@ -562,20 +562,24 @@ test('can get account transfers', async (): Promise<void> => {
...
   } finally {
     await client.destroy()
   }
+
+  const end = new Date().getTime()
+  console.log('Time taken (s):', (end - start)/1000)  
+  process.exit(fail_count);
 }
 
 main().catch((error: AssertionError) => {
```
Comment (matklad) [inline src/clients/node/src/test.ts:581]: Please rebase away changes to the lockfile, we intend to keep it frozen until we remove docusaurus altogether. 

(I tried commented on the lockfile diff, but my browser crashed)

Comment (matklad) [inline src/clients/node/src/test.ts:581]: Let’s rather exit process in the .catch to main here, to have a guarantee that any exception results in abnormal termination

File: src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java (hunk 1 above)
Comment (batiati) [reply src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java]: TIL it's possible to have more than one object inside a Java try-resource block!!

> And I think we don’t need catch here?

Yes, since we don't throw on `close` anymore, we can remove this catch block!

**Review by matklad — DISMISSED** [review]

**Comment by matklad** [comment]
Failure seems legit:

```
    Failed CreatePendingTransfersAndVoidExpiredAsync [6 s]
    Error Message:
     Assert.IsTrue failed. 
    Stack Trace:
       at TigerBeetle.Tests.IntegrationTests.CreatePendingTransfersAndVoidExpiredAsync() in /home/runner/work/tigerbeetle/tigerbeetle/src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:line 723
```

**Comment by batiati** [comment]
> Failure seems legit:

Looks like `Task.Delay` does not have accurate resolution and sometimes finishes sooner than the timeout 😨.
Kind of the same problem here: https://github.com/zkSNACKs/WalletWasabi/issues/6166

**Review by matklad — APPROVED** [review]
